File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/java/Jumpif.java
public class Jumpif extends Instruccion{

    private int c;

    public Jumpif(int c){
        this.c = c;
    }

    public void ejecutar(Programa programa){
        int h = programa.getPila().pop();
        if (h >= 0){
            programa.salto(c);
        }else{
            programa.inc();
        }
    }

    public String mostrar(){
        return "jumpif " + c;
    }
}

File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/c++/jumpif.cc
#include "jumpif.h"
using namespace std;

Jumpif::Jumpif(int c) : c(c){}

void Jumpif::ejecutar(Programa* programa){
    int h = programa->getPila().top();
    programa->getPila().pop();
    if (h >= 0){
        programa->salto(c);
    }else{
        programa->inc();
    }
}

string Jumpif::mostrar(){
    return("jumpif " + to_string(c));
}
